src/main/scala/org/apache/spark/sql/redis/RedisSourceRelation.scala

```diff
@@ -112,14 +116,14 @@ class RedisSourceRelation(override val sqlContext: SQLContext,
     }
 
     // write data
-    data.foreachPartition { partition =>
+    val writeData = if (numPartitions > 0) data.repartition(numPartitions) else data
```

Comment:
'numPartitions' option should not apply to writes, so we shouldn't do any repartition while saving to Redis. It's up to the user to create a dataframe with a certain number of partitions before saving.